### PR TITLE
Add local-ui-dev mode and fix rancher ui expose registry

### PR DESCRIPTION
**Pull request by guangbochen**
**Problem:**
- [BUG] Rancher enabled settings does not enable the Rancher button in the header #1133 
- [BUG]Rancher UI on first screen when Installing Harvester using IPXE Boot #1139 

**Solution:**
- remove deprecated settings 
- add local UI dev mode with set `--add-local-ui=true` to enable local UI for local development
- add rancher-expose svc after patched rancher settings on the Harvester server.

**Related Issue:**
https://github.com/harvester/harvester/issues/1133
https://github.com/harvester/harvester/issues/1139

**Test plan:**

### CODE REVIEW

File: pkg/server/router.go [hunk 1]
```diff
@@ -75,27 +74,30 @@ func (r *Router) Routes(h router.Handlers) http.Handler {
...
-		if err != nil {
-			logrus.Fatal(err)
-		}
-		rancherHandler := &proxy.Handler{
-			Host:   host,
-			Scheme: scheme,
+	// enable local dashboard ui for dev
+	if r.options.AddLocalUI {
+		loginHandler := auth.NewLoginHandler()
+		m.Path("/v1-public/auth").Handler(loginHandler)
```
Comment (gitlawr) [inline pkg/server/router.go]: I thought we removed the auth part. Do we still need this?

File: main.go [hunk 2]
```diff
@@ -76,6 +76,12 @@ func main() {
 			Usage:       "Specify the URL to connect to the Rancher server",
 			Destination: &options.RancherURL,
 		},
+		cli.BoolFlag{
+			Name:        "add-local-ui",
+			Usage:       "Add local dashboard UI(true, false)",
```
Comment (bk201) [inline main.go]:
```suggestion
			Usage:       "Add local dashboard UI (true, false)",
```

nit.

**Comment by guangbochen** [comment]
Updated, can u guys PTAL thanks.

Additionally add fix of invalid feature gates config, previous the feature gates of Kubevirt only contains the `LiveMigration`, but `HotplugVolumes` is also required.
```helm template deploy/charts/harvester | grep "KubeVirt" --context=30```

**Review by gitlawr — APPROVED** [review]
LGTM

File: pkg/server/router.go [hunk 3]
```diff
@@ -75,27 +73,27 @@ func (r *Router) Routes(h router.Handlers) http.Handler {
...
-			Host:   host,
-			Scheme: scheme,
+	// enable local dashboard ui for dev
+	if r.options.AddLocalUI {
+		vueUI := ui.Vue
+		m.Handle("/dashboard/", vueUI.IndexFile())
+		m.PathPrefix("/dashboard/").Handler(vueUI.IndexFileOnNotFound())
+		m.PathPrefix("/api-ui").Handler(vueUI.ServeAsset())
+
+		if r.options.AddLocalUI || r.options.RancherURL != "" {
```
Comment (bk201) [inline pkg/server/router.go]: Everything is good except the first condition is not needed.

**Review by bk201 — APPROVED** [review]

File: pkg/server/router.go (hunk 3 above)
Comment (guangbochen) [reply pkg/server/router.go]: fixed, thanks

**Pull request merged**